tfjs-backend-webgpu/src/kernels/matmul_small_output_size_webgpu.ts

```diff
@@ -107,29 +108,130 @@ export function makeMatMulSmallOutputSizeSource(): string {
   `;
 }
 
+export function makeMatMulSmallOutputSizeSourceWgsl(
+    workGroupSize: [number, number, number]): string {
+  const tileAOuter = workGroupSize[1] / 2;
+  const tileBOuter = workGroupSize[0];
+  const tileInner = tileAOuter > tileBOuter ? tileAOuter : tileBOuter;
+  return `
+  var<workgroup> mm_Asub1 : array<array<f32, ${tileInner}>, ${tileAOuter}>;
+  var<workgroup> mm_Bsub1 : array<array<f32, ${tileBOuter}>, ${tileInner}>;
+  var<workgroup> mm_Asub2 : array<array<f32, ${tileInner}>, ${tileAOuter}>;
+  var<workgroup> mm_Bsub2 : array<array<f32, ${tileBOuter}>, ${tileInner}>;
+
+  // If the output size is small for matrix multiplication, avoid to use vec4
+  // and handle some elements per thread to optimally utilize the ALU.
+  // Introduces two shared memory buffers, some logical threads could handle
+  // arithmetic operations and others handle IO operations between barrier api,
+  // makes ALUs and load/store units work simultaneously, could improves
+  // the performance.
+  ${getMainHeaderStringWgsl(workGroupSize)} {
+    ${getGlobalIndexStringWgsl(workGroupSize)}
```

Comment:
nit: Why is `getGlobalIndexStringWgsl` needed?

Reply:
You are right, it is not used, removed.